### PR TITLE
storage concepts documentation

### PR DESCRIPTION
**Pull request by agavra**

### CODE REVIEW

File: docs/concepts/storage.md [hunk 1]
```diff
@@ -3,3 +3,157 @@ sidebar_position: 3
...
+   Streams deployments in the wild tend to overprovision on both to ensure
+   that neither limitation is met.
+1. Scaling out becomes difficult since new nodes need to bootstrap storage
+   before they can become operational.
+1. Recovering from a fault or rebalancing is a challenge for the same reason
+   as above.
+
+:::info
+
+Curious to read more about tuning RocksDB state stores and contrasting it with
```
Comment (apurvam) [inline docs/concepts/storage.md:38]: Let's drop this until we publish the blog. At launch the link will be broken.

File: docs/concepts/storage.md [hunk 2]
```diff
@@ -3,3 +3,157 @@ sidebar_position: 3
...
+to these problems: standby and warmup replicas, incremental rebalancing, sticky 
+task assignment... the list goes on.  Responsive takes a different approach to 
+solving this class of problems in Kafka Streams by improving the underlying 
+architecture to separate compute and storage - effectively eliminating these 
+concerns in one fell swoop.
+
+## Remote Storage
+
+
+At Responsive, we believe that reactive backends should power the most business-
```
Comment (apurvam) [inline docs/concepts/storage.md:54]: I feel we can drop the marketing angle from the doc. We should just say "We've built our storage solution to leverage ScyllaDB, which is a battle tested distributed database that provides: ..".

File: docs/concepts/storage.md [hunk 3]
```diff
@@ -3,3 +3,157 @@ sidebar_position: 3
...
+  collocating more partitions on a single node while RocksDB based solutions
+  start to degrade.
+
+## Additional Features
+
+Beyond providing functional parity with RocksDB, Responsive remote stores
+promise the ability to implement features that were not easy to implement
+with a RocksDB based solution.
+
+Already, Responsive offers time-to-live (or `ttl`) based on wall-clock - one
```
Comment (apurvam) [inline docs/concepts/storage.md:147]: I think the paragraph on the TTL store and IQ can have their own subheadings.

File: docs/concepts/storage.md [hunk 4]
```diff
@@ -3,3 +3,157 @@ sidebar_position: 3
...
+## Remote Storage
+
+
+At Responsive, we believe that reactive backends should power the most business-
+critical applications in your stack, that's why we built our storage solution
+to leverage [Scylla DB](https://github.com/scylladb/scylladb), a battle-tested
+distributed database that provides:
+
+- Fast writes & fast Key-Value reads
+- The transactional guarantees to implement the EOS `processing.guarantee`
```
Comment (apurvam) [inline docs/concepts/storage.md:60]: should we spell out `processing.guarantee=exactly_once_v2`?

File: docs/concepts/storage.md [hunk 5]
```diff
@@ -3,3 +3,157 @@ sidebar_position: 3
...
+The motivation for such a design was clear at the time: most deployments of
+Kafka Streams were running without access to the public cloud, and keeping
+the library free of external dependencies other than Kafka made it uniquely
+easy to deploy.
+
+The limitations of copuling compute with storage are well known: 
+1. Sizing your nodes becomes more of an art than a science, and many Kafka
+   Streams deployments in the wild tend to overprovision on both to ensure
+   that neither limitation is met.
+1. Scaling out becomes difficult since new nodes need to bootstrap storage
```
Comment (apurvam) [inline docs/concepts/storage.md:31]: 'You don't have elastic scaleout since ..'

File: docs/concepts/storage.md (hunk 2 above)
Comment (apurvam) [reply docs/concepts/storage.md:54]: I think we should also mention that with managed services in the cloud, the motivation for the original design is somewhat invalidated, creating an opening for our solution.

File: docs/concepts/storage.md [hunk 6]
```diff
@@ -3,3 +3,157 @@ sidebar_position: 3
...
+
+:::tip TLDR
+
+The punchline from this concepts doc is that Responsive Platform separates
+storage from compute as a means of improving operations without sacraficing
+transactional semantics at a neglible performance penalty.
+
+:::
+
+## History & Background
```
Comment (apurvam) [inline docs/concepts/storage.md:15]: General comment: I feel we are spending relatively more time on the old architecture than _how_ our solution works. Maybe there isn't much to say on the latter, but this is the main high level thought I had reading this.

File: docs/concepts/storage.md [hunk 7]
```diff
@@ -3,3 +3,157 @@ sidebar_position: 3
...
+next few sections examine some of the concepts we've implemented.
+
+### Instant Restoration
+
+With a RocksDB-based deployment of Kafka Streams, restarting a node that has
+both a RocksDB store and a checkpoint file is easy: the application will know
+where from the changelog to start reading from and skip to that point before
+starting restoration.
+
+Responsive's remote storage takes inspiration from that strategy, and makes
```
Comment (apurvam) [inline docs/concepts/storage.md:76]: I think we can go into more detail on how we use the changelog to leverage kafka transactions and exactly once, while not requiring long restore times.

I also think we should talk about how we have managed decoupling the partitioning for storage from the partitioning of the kafka topics, which should allow easier repartitioning of Kafka topics.

**Review by apurvam — APPROVED** [review]
LGTM, I left some thoughts, but I think this is good to merge after removing the link to the non-existent blog.

File: docs/concepts/storage.md (hunk 6 above)
Comment (agavra) [reply docs/concepts/storage.md:15]: That's fair, though also somewhat intentional - I didn't want this page to become quickly out of sync with what's actually deployed and I also wanted to focus on _what_ we provide (especially in contrast to what vanilla KStreams provides) as opposed to how we accomplish that (dropping hints to the latter) as I think that's what's more interesting from a documentation perspective.

File: docs/concepts/storage.md (hunk 7 above)
Comment (agavra) [reply docs/concepts/storage.md:76]: I'll add a section on the latter - that's a great call, the former is pretty deep and I think if we wanted we could just write an entire section or a blog about it - I don't think it makes sense to go into detail of our EOS design here.

**Pull request merged**